headers/validation.h

```diff
@@ -108,6 +109,52 @@ auto operator|(T1 /*unused*/, T2 /*unused*/) {
 
 enum Separator { Space, Newline };
 
+//this contains some specific code which would be available in c++20
```

Comment:
I read this as 'this depends on c++20' at first. Maybe make it slightly clearer that instead this would be replaced by c++20.